servers/physics_2d/area_2d_sw.cpp

```diff
@@ -215,7 +215,9 @@ void Area2DSW::call_queries() {
 
 		for (Map<BodyKey, BodyState>::Element *E = monitored_bodies.front(); E;) {
```

Comment:
Just to understand the logic for me here, the previous commit mentioned that you wanted to handle case where new monitored bodies/areas are added inside the callbacks. I'm not sure how various things in the engine work yet, so just want to learn something.

When new bodies are added while iteration is happening, do you expect them to be processed in the same iteration or next one? I.e. do you expect monitored_bodies to have elements remaining in case callbacks add more monitored bodies?

Reply:
It depends on the order they're processed in. If the body or area has already been processed, it will be processed again next time, if not, it will be processed in this iteration. The previous problem was, when they had already been processed, they were not being processed on the next iteration.